### PR TITLE
More flexible obs space in cycling scripts

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description

Updates to the cycling scripts. 
- Added GEOS forecast model
- Switching the obsspace setup in the 3dvar yaml to do something similar to what we have for EMC
- Updates to how we handle `input.nml`

### Issue(s) addressed

We should be able to re-use what we developed for EMC in terms of the ObsSpace setup, which is done per instrument, as opposed to general obs types (e.g. gmi.sst instead of sst ... etc)

## Testing

Tested on discover and linux workstations. This code is not part of our automated travis-ci tests.
These are very simple and easily portable scripts, I will add the necessary data to allow us to use them on Hera and Orion.

## Dependencies

[soca-config pr#20](https://github.com/JCSDA/soca-config/pull/20)

### CODE REVIEW

File: scripts/workflow/apps/exp.config.mom6solo [hunk 1]
```diff
@@ -0,0 +1,52 @@
+EXP_DIR=$(readlink -f .)
```
Comment (aerorahul) [inline scripts/workflow/apps/exp.config.mom6solo:1]: This is specific to GEOS and Discover.
The filename should reflect that. It is not generic.

Comment (aerorahul) [reply scripts/workflow/apps/exp.config.mom6solo:1]: it also contains specific references to some one named gvernier and userspace.
Double no-no.

Comment (guillaumevernieres) [reply scripts/workflow/apps/exp.config.mom6solo:1]: @aerorahul exp.config.discover.mom6solo?

Comment (guillaumevernieres) [reply scripts/workflow/apps/exp.config.mom6solo:1]: > it also contains specific references to some one named gvernier and userspace.
> Double no-no.

great guy so I've heard.

Comment (aerorahul) [reply scripts/workflow/apps/exp.config.mom6solo:1]: > @aerorahul exp.config.discover.mom6solo?
It runs GEOS via gcm_run.j so it is not mom6solo is it?

**Review by aerorahul — APPROVED** [review]
much better.
Alas, I can't test on Discover. Something to do with my desertion.
I will try and cycle on my desktop.

**Pull request merged**